js/src/main/scala/scalatags/jsdom/Tags.scala

```diff
@@ -3,94 +3,95 @@ package scalatags
-  val base = "base".voidTag[dom.HTMLBaseElement]
-  val link = "link".voidTag[dom.HTMLLinkElement]
-  val meta = "meta".voidTag[dom.HTMLMetaElement]
+  val head = "head".tag[Head]
+  val base = "base".voidTag[Base]
+  val link = "link".voidTag[Link]
+  val meta = "meta".voidTag[Meta]
   // Scripting
-  val script = "script".tag[dom.HTMLScriptElement]
+  val script = "script".tag[Script]
   // Sections
-  val body = "body".tag[dom.HTMLBodyElement]
-  val h1 = "h1".tag[dom.HTMLHeadingElement]
-  val h2 = "h2".tag[dom.HTMLHeadingElement]
-  val h3 = "h3".tag[dom.HTMLHeadingElement]
-  val h4 = "h4".tag[dom.HTMLHeadingElement]
-  val h5 = "h5".tag[dom.HTMLHeadingElement]
-  val h6 = "h6".tag[dom.HTMLHeadingElement]
-  val header = "header".tag[dom.HTMLElement]
-  val footer = "footer".tag[dom.HTMLElement]
+  val body = "body".tag[Body]
+  val h1 = "h1".tag[Heading]
+  val h2 = "h2".tag[Heading]
+  val h3 = "h3".tag[Heading]
+  val h4 = "h4".tag[Heading]
+  val h5 = "h5".tag[Heading]
+  val h6 = "h6".tag[Heading]
+  val header = "header".tag[Html]
+  val footer = "footer".tag[Html]
   // Grouping content
-  val p = "p".tag[dom.HTMLParagraphElement]
-  val hr = "hr".voidTag[dom.HTMLHRElement]
-  val pre = "pre".tag[dom.HTMLPreElement]
-  val blockquote = "blockquote".tag[dom.HTMLQuoteElement]
-  val ol = "ol".tag[dom.HTMLOListElement]
-  val ul = "ul".tag[dom.HTMLUListElement]
-  val li = "li".tag[dom.HTMLLIElement]
-  val dl = "dl".tag[dom.HTMLDListElement]
-  val dt = "dt".tag[dom.HTMLDTElement]
-  val dd = "dd".tag[dom.HTMLDDElement]
-  val figure = "figure".tag[dom.HTMLElement]
-  val figcaption = "figcaption".tag[dom.HTMLElement]
-  val div = "div".tag[dom.HTMLDivElement]
+  val p = "p".tag[Paragraph]
+  val hr = "hr".voidTag[HR]
+  val pre = "pre".tag[Pre]
+  val blockquote = "blockquote".tag[Quote]
+  val ol = "ol".tag[OList]
+  val ul = "ul".tag[UList]
+  val li = "li".tag[LI]
+  val dl = "dl".tag[DList]
+  val dt = "dt".tag[DT]
+  val dd = "dd".tag[DD]
+  val figure = "figure".tag[Html]
+  val figcaption = "figcaption".tag[Html]
+  val div = "div".tag[Div]
   // Text-level semantics
-  val a = "a".tag[dom.HTMLAnchorElement]
-  val em = "em".tag[dom.HTMLElement]
-  val strong = "strong".tag[dom.HTMLElement]
-  val small = "small".tag[dom.HTMLElement]
-  val s = "s".tag[dom.HTMLElement]
-  val cite = "cite".tag[dom.HTMLElement]
-  val code = "code".tag[dom.HTMLElement]
-  val sub = "sub".tag[dom.HTMLElement]
-  val sup = "sup".tag[dom.HTMLElement]
-  val i = "i".tag[dom.HTMLElement]
-  val b = "b".tag[dom.HTMLElement]
-  val u = "u".tag[dom.HTMLElement]
-  val span = "span".tag[dom.HTMLSpanElement]
-  val br = "br".voidTag[dom.HTMLBRElement]
-  val wbr = "wbr".voidTag[dom.HTMLElement]
+  val a = "a".tag[Anchor]
+  val em = "em".tag[Html]
+  val strong = "strong".tag[Html]
+  val small = "small".tag[Html]
+  val s = "s".tag[Html]
+  val cite = "cite".tag[Html]
+  val code = "code".tag[Html]
+  val sub = "sub".tag[Html]
```

Comment:
Similarly these should be `html.Element` (`dom.HTMLElement`) rather than `dom.HTMLHTMLElement`